### PR TITLE
Add Deals Model and DB migration

### PR DESCRIPTION
**Pull request by ahmad-asal**

### CODE REVIEW

File: src/database/models/deal.model.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            },
+            expiryDate: {
+                type: DataTypes.DATE,
+                allowNull: true,
+            },
+            status: {
+                type: DataTypes.ENUM('In Review', 'Approved', 'Rejected'),
+                allowNull: false,
+            },
+        },
```
Comment (DuaaH) [inline src/database/models/deal.model.ts:58]: I think we need to add a userId foreign key to the DealModel.
and for the deals, we need to upload images so we need a column for the images.

Comment (ahmad-asal) [reply src/database/models/deal.model.ts:58]: @DuaaH Yes, these will be handled in different PR, this is the started PR for the deals

File: src/database/models/index.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import { DealModel } from './deal.model';
+
+// Define associations
+DealModel.belongsTo(CategoryModel, {
+    foreignKey: 'categoryId',
+    as: 'category',
+});
+CategoryModel.hasMany(DealModel, { foreignKey: 'categoryId' });
+
+export { CategoryModel, DealModel };
```
Comment (muradrajab) [inline src/database/models/index.ts:13]: it seems you mistakenly exporting deal model

Comment (ahmad-asal) [reply src/database/models/index.ts:13]: In this module, we are going to export all models with associations to avoid circular dependencies

Comment (muradrajab) [reply src/database/models/index.ts:13]: can you explain more how circuler dependency will happen in this can and it's implications.

Comment (muradrajab) [reply src/database/models/index.ts:13]: i understand .
but this problem came after you introduce the index file to manage **all applications associations** inside of it.,
which i think  will make it mess to. in contrast to   _setting  association in either or both of it's parties ( in this case either inside deal and/or categor ) will be much more mangable_

Comment (ahmad-asal) [reply src/database/models/index.ts:13]: As discussed, we have multiple approaches for such problems, already discussed in the meeting, closing the thread

**Pull request merged**